pymagicc/io.py

```diff
@@ -611,6 +654,145 @@ def _read_notes(self):
         return notes
 
 
+class _OutReader(_FourBoxReader):
+    _regexp_capture_variable = re.compile(r"DAT\_(\w*)\.OUT$")
+    _default_todo_fill_value = "N/A"
+
+
+class _TempOceanLayersOutReader(_InputReader):
+    _regexp_capture_variable = re.compile(r"(TEMP\_OCEANLAYERS\_?\w*)\.OUT$")
+    _default_todo_fill_value = "N/A"
+
+    def _read_magicc6_style_header(self, stream, metadata):
+        column_headers, metadata = super()._read_magicc6_style_header(stream, metadata)
+
+        if set(column_headers["variables"]) == {"TEMP_OCEANLAYERS"}:
+            region = "GLOBAL"
+        elif set(column_headers["variables"]) == {"TEMP_OCEANLAYERS_NH"}:
+            region = "NHOCEAN"
+        elif set(column_headers["variables"]) == {"TEMP_OCEANLAYERS_SH"}:
+            region = "SHOCEAN"
+        else:
+            self._raise_cannot_determine_variable_from_filename_error()
+
+        column_headers["variables"] = [
+            "OCEAN_TEMP_" + l for l in column_headers["regions"]
+        ]
+        column_headers["regions"] = [region] * len(column_headers["regions"])
+
+        return column_headers, metadata
+
+
+class BinData(object):
```

Comment:
why is this public?

Reply:
Everything in Python is public?? I'll refactor the name to `_BinData`.

Reply:
sorry you know what I mean, thanks